### DISCUSSION TITLE
Support extracting classes like tailwind @apply

### DISCUSSION THREAD

**Discussion by khanh-le-otsv** [chakra-ui/panda#1153] [Ideas]
TailwindCSS has a feature that supports extracting styles to classes: `https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply`. Supporting this feature would be great.

**Comment by anubra266** [comment]
@khanh-le-otsv This might be helpful
https://panda-css.com/docs/guides/dynamic-styling#using-static-css

**Comment by khanh-le-otsv** [comment]
I suggest this feature because panda sometimes generates too many classes, it's good if I can combine them into one

**Reply by anubra266** [reply]
@khanh-le-otsv Panda classes are not constant like tailwind, they're very dynamic and customizable, so that's not feasible for now.
But if long classNames is what you're worried about, you can solve this with `recipes`, `textStyles` and `layerStyles`
